src/support/insert_ordered.h

```diff
@@ -126,13 +140,26 @@ template<typename Key, typename T> struct InsertOrderedMap {
   size_t count(const Key& k) const { return Map.count(k); }
 
   InsertOrderedMap() = default;
-  InsertOrderedMap(InsertOrderedMap& other) {
-    // TODO, watch out for iterators.
-    WASM_UNREACHABLE("unimp");
+  InsertOrderedMap(const InsertOrderedMap& other) {
+    for (auto kv : other) {
+      insert(kv);
+    }
   }
   InsertOrderedMap& operator=(const InsertOrderedMap& other) {
-    // TODO, watch out for iterators.
-    WASM_UNREACHABLE("unimp");
+    if (this != &other) {
+      this->~InsertOrderedMap();
+      new (this) InsertOrderedMap<Key, T>(other);
+    }
+    return *this;
+  }
+  InsertOrderedMap(InsertOrderedMap&& other)
+    : Map(std::move(other.Map)), List(std::move(other.List)) {}
```

Comment:
Are move semantics guaranteed to get this right? I feel like I've read that they have, but I've never depended on that as much as this line seems to...

Reply:
Good question. I couldn't find any specific spec guarantees, so I removed this new constructor and assignment operator. They were left over from shots in the dark trying to fix a bug that turned out to be unrelated.